dev/reference/tools/tidb-lightning/glossary.md

```diff
@@ -0,0 +1,187 @@
+
+### Chunk
+
+Equivalent to a single file in the data source.
+
+### Compaction
+
+An operation which merges multiple small SST files into a large SST file, and cleans up deleted entries.
+
+TiKV automatically compacts data in background while TiDB Lightning is importing. Due to legacy reasons, you can configure TiDB Lightning to explicitly trigger a compaction every time a table is imported. This is now discouraged and the corresponding settings are disabled by default.
+
+See [RocksDB's wiki page on Compaction](https://github.com/facebook/rocksdb/wiki/Compaction) for its technical details.
+
+<!-- D -->
+
+## D
+
+### Data engine
+
+An [engine](/dev/reference/tools/tidb-lightning/glossary.md#engine) for sorting actual row data.
+
+When a table is very large, its data is placed into multiple data engines to improve task pipelining and save space of TiKV Importer. By default a new data engine is opened every 100 GB of SQL data, which can be configured through the `mydumper.batch-size` setting.
+
+TiDB Lightning processes multiple data engines concurrently. This is controlled by the `lightning.table-concurrency` setting.
+
+<!-- E -->
+
+## E
+
+### Engine
+
+In TiKV Importer, an engine is a RocksDB instance for sorting KV pairs.
+
+TiDB Lightning transfers data to TiKV Importer through engines. It first opens an engine, sends lots of KV pairs to it (with no particular order), and finally closes the engine. The engine will sort the received KV pairs after it is closed. These closed engines can then be further uploaded to the TiKV stores for ingestion.
+
+Engines use TiKV Importer's `import-dir` as temporary storage, which are sometimes referred as "engine files".
+
+See also [data engine](/dev/reference/tools/tidb-lightning/glossary.md#data-engine) and [index engine](/dev/reference/tools/tidb-lightning/glossary.md#index-engine).
+
+<!-- I -->
+
+## I
+
+### Import mode
+
+A configuration which optimizes TiKV for writing but reduces read speed and space utility.
+
+TiDB Lightning automatically switches into and out of import mode while running. However, if TiKV is somehow stuck in import mode, you can use `tidb-lightning-ctl` to [force them revert](/dev/faq/tidb-lightning.md#why-my-tidb-cluster-is-using-lots-of-cpu-resources-and-running-very-slowly-after-using-tidb-lightning) to [normal mode](/dev/reference/tools/tidb-lightning/glossary.md#normal-mode).
+
+### Index engine
+
+An [engine](/dev/reference/tools/tidb-lightning/glossary.md#engine) for sorting indices.
+
+Regardless of number of indices, every table is associated with exactly one index engine.
+
+TiDB Lightning processes multiple index engines concurrently. This is controlled by the `lightning.index-concurrency` setting. Since every table has exactly one index engine, this also configures the maximum number of tables to process in at the same time.
```

Comment:
```suggestion
TiDB Lightning processes multiple index engines concurrently. This is controlled by the `lightning.index-concurrency` setting. Since every table has exactly one index engine, this also configures the maximum number of tables to process at the same time.
```